### PR TITLE
[controller] Add DrbdStorageClassWatcher to reconcile DrbdStorageClass.

### PR DESCRIPTION
**Pull request by ViktorKram**

### CODE REVIEW

File: images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+					setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				} else {
+					removeNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				}
+			}
+		default:
+			err := errors.New("unsupported replication type")
+			log.Error(err, fmt.Sprintf("[ReconcileDRBDStorageClassReplication] replication type validation failed for DRBDStorageClass %s", dsc.Name))
+
+			setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, "storage.deckhouse.io/non-operational")
```
Comment (AleksZimin) [inline images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go:194]: Why not ?
```suggestion
			setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
```

File: images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+	ctx context.Context,
+	cl client.Client,
+	log logger.Logger,
+	dscs map[string]v1alpha1.DRBDStorageClass,
+	sps map[string][]lapi.StoragePool,
+) map[string]v1alpha1.DRBDStorageClass {
+	healthy := make(map[string]v1alpha1.DRBDStorageClass, len(dscs))
+	for _, dsc := range dscs {
+		if _, exist := sps[dsc.Spec.StoragePool]; exist {
+			healthy[dsc.Name] = dsc
```
Comment (AleksZimin) [inline images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go:109]: I think we need this one
```suggestion
			healthy[dsc.Name] = dsc
			removeNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByStoragePool)
```

File: images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go (hunk 1 above)
Comment (ViktorKram) [reply images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go:194]: The label says there are not enough nodes in the selected zones, but in the default case we face an unsupported replication type, so the message about "not enough zones" is not relevant in this case. 
As long as the case with unsupported replication type is about to never happens, I decided not to create a special label but use some general one instead.

File: images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go (hunk 2 above)
Comment (ViktorKram) [reply images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go:109]: Yes, true. Thanks.

**Review by AleksZimin — APPROVED** [review]

**Pull request merged**